### PR TITLE
RELEASE_v2.4.0-test

### PR DESCRIPTION
**Pull request by MinhLA1410**
Changes:
- support PostgreSQL 12.16, 13.12, 14.9, 15.4 and 16.0
- synchronize test code between versions.

### CODE REVIEW

File: expected/13.12/extra/encodings.out [hunk 1]
```diff
@@ -34,6 +34,41 @@
...
+ epo | Laŭ Ludoviko Zamenhof bongustas freŝa ĉeĥa manĝaĵo kun spicoj.
+ cze | Zvlášť zákeřný učeň s ďolíčky běží podél zóny úlů.
+ ara | أبجد هوَّز حُطّي كلَمُن سَعْفَص قُرِشَت ثَخَدٌ ضَظَغ
+ heb | עטלף אבק נס דרך מזגן שהתפוצץ כי חם
+(20 rows)
+
+DROP FOREIGN TABLE "Unicode data";
+DROP SERVER sqlite_svr;
+DROP EXTENSION sqlite_fdw;
 -- euc_jp
```
Comment (mkgrgis) [inline expected/13.12/extra/encodings.out:72]: This is `encodings` test artefacts moved to encodings.sql. Also this is applicable to other versions.

**Comment by mkgrgis** [comment]
@MinhLA1410 , please refer https://github.com/pgspider/sqlite_fdw/compare/master...mkgrgis:sqlite_fdw:fix_uuid_and_tests?expand=1 file changes where I have unify `uuid`, `bool`, `bitstring` and `encodings` tests. This is draft branch for PR after https://github.com/pgspider/sqlite_fdw/pull/88 and https://github.com/pgspider/sqlite_fdw/pull/91. You have got `encodings` test artefacts. I am sorry, my test normalisation was planned later than this your PR.

**Comment by t-kataym** [comment]
@mkgrgis Please let me merge the PR #93 first. You are familiar with refactoring the test code, could you rebase #88 and #91? I'm sorry for bothering you.

**Comment by mkgrgis** [comment]
@t-kataym , no problems. I have scripts to unify tests between versions.

Also I think it's important to delete `encodings` test artefacts in **this** PR. In #88 also there is

> Remove `encoding`s test artefacts from sqlite_fdw main test with PG 15 and 16.

Hence rebasing of #88 will be more productive for me in case of @MinhLA1410 will delete this artefacts now. It is yet results of last unsuccessful test rebasing with unification and version updating.

**Comment by MinhLA1410** [comment]
@mkgrgis, Do you want to remove this section? (https://github.com/pgspider/sqlite_fdw/pull/93/files#diff-d52fef6eb88d3a0df647c84e6a43d1f0cc573269296110d6207f22d7edcc13e5R38)

File: sql/16.0/sqlite_fdw.sql [hunk 2]
```diff
@@ -782,3610 +786,9 @@ DROP FOREIGN TABLE grem1_1;
...
 DROP FOREIGN TABLE case_exp;
---test for PR #76 github
-DROP FOREIGN TABLE "Unicode data";
 
 --Testcase 151:
 DROP SERVER sqlite_svr;
 --Testcase 152:
 DROP EXTENSION sqlite_fdw CASCADE;
 
--- tests for PR #76 github
```
Comment (mkgrgis) [inline sql/16.0/sqlite_fdw.sql:793]: Yet deleted encodings test artefacts, no problems.

**Comment by mkgrgis** [comment]
> @mkgrgis, Do you want to remove this section? (https://github.com/pgspider/sqlite_fdw/pull/93/files#diff-d52fef6eb88d3a0df647c84e6a43d1f0cc573269296110d6207f22d7edcc13e5R38)

No, @MinhLA1410 . Please refer a full `encodings` test files https://github.com/mkgrgis/sqlite_fdw/blob/fix_uuid_and_tests/expected/16.0/extra/encodings.out and https://github.com/mkgrgis/sqlite_fdw/blob/fix_uuid_and_tests/sql/16.0/extra/encodings.sql . In your PR this files is also in normal form, no problems. Also I see you have deleted doubled test artifacts  in [sql/16.0/sqlite_fdw.sql](https://github.com/pgspider/sqlite_fdw/pull/93/files/713b47bc8ae1d071c3989344bb5d6d0a76f99104#diff-025b87396f4f252f6fdb4561ef4d3f8457adee8ca55f9d3fc8ce20b885c10dbe) hence no problems. I am sorry, this PR is OK. Let's merge.

File: expected/13.12/extra/encodings.out (hunk 1 above)
Comment (mkgrgis) [reply expected/13.12/extra/encodings.out:72]: I am sorry. This is not `encodings` test artifacts, `SELECT * FROM "Unicode data";` also was moved to this test. Thanks, @MinhLA1410 ! I haven't expect this optimization. All is ok.

**Comment by t-kataym** [comment]
@mkgrgis , @MinhLA1410 Thank you for confirming the PR and discussing.

**Pull request merged**